src/virtualenv/create/via_global_ref/builtin/python2/site.py

```diff
@@ -13,8 +13,14 @@ def main():
     sys.real_prefix = sys.base_prefix = config["base-prefix"]
     sys.base_exec_prefix = config["base-exec-prefix"]
     sys.base_executable = config["base-executable"]
+    base_executable = abs_path(sys.base_executable)
+    base_prefix, base_exec_prefix = abs_path(sys.base_prefix), abs_path(sys.base_exec_prefix)
+    exe = abs_path(sys.executable)
+    exe_dir = exe[: exe.rfind(sep)]
+    prefix, exec_prefix = abs_path(sys.prefix), abs_path(sys.exec_prefix)
     global_site_package_enabled = config.get("include-system-site-packages", False) == "true"
-    rewrite_standard_library_sys_path()
+    rewrite_standard_library_sys_path(base_executable, exe_dir, exec_prefix, base_prefix, prefix, base_exec_prefix)
```

Comment:
Feels to me we only need all these variables here because we're doing the rewrite and revert in two functions, cant' we just do it in one to avoid this extra overhead?

Reply:
Sure